expression/column.go

```diff
@@ -38,10 +38,9 @@ type CorrelatedColumn struct {
 
 // Clone implements Expression interface.
 func (col *CorrelatedColumn) Clone() Expression {
-	var d types.Datum
 	return &CorrelatedColumn{
 		Column: col.Column,
-		Data:   &d,
+		Data:   col.Data,
```

Comment:
this change may cause the data race in `ParallelNestedLoopApplyExec`, which should fix later